src/tcache.c

```diff
@@ -958,6 +980,54 @@ thread_tcache_max_set(tsd_t *tsd, size_t tcache_max) {
 	assert(tcache_nbins_get(tcache_slow) == sz_size2index(tcache_max) + 1);
 }
 
+bool
+tcache_bins_ncached_max_write(tsd_t *tsd, char *settings, size_t len) {
+	tcache_t *tcache = tsd_tcachep_get(tsd);
+	bool enabled = tcache_available(tsd);
+	assert(tcache != NULL);
+	cache_bin_info_t tcache_bin_info[TCACHE_NBINS_MAX];
+	tcache_bin_settings_backup(tcache, tcache_bin_info);
+	const char *bin_settings_segment_cur = settings;
+	size_t len_left = len;
+
```

Comment:
The input len could be 0. Lets make the call site not calling down with 0 len and assert here.

Reply:
If the input length is 0, should we recognize that as `EINVAL` or simply return without doing anything?

Reply:
I'd say do nothing and return. Actually we should test more of the bad inputs, e.g. half finished string, string longer than the 1k limit.

Reply:
Sure. Added some tests with invalid inputs